src/write.c

```diff
@@ -899,81 +996,87 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
     // Write iprp -> ipco/ipma
 
     avifBoxMarker iprp = avifRWStreamWriteBox(&s, "iprp", AVIF_BOX_SIZE_TBD);
-
-    uint8_t itemPropertyIndex = 0;
-    avifBoxMarker ipco = avifRWStreamWriteBox(&s, "ipco", AVIF_BOX_SIZE_TBD);
-    for (uint32_t itemIndex = 0; itemIndex < encoder->data->items.count; ++itemIndex) {
-        avifEncoderItem * item = &encoder->data->items.item[itemIndex];
-        const avifBool isGrid = (item->gridCols > 0);
-        memset(&item->ipma, 0, sizeof(item->ipma));
-        if (!item->codec && !isGrid) {
-            // No ipma to write for this item
-            continue;
-        }
-
-        if (item->dimgFromID) {
-            // All image cells from a grid should share the exact same properties, so see if we've
-            // already written properties out for another cell in this grid, and if so, just steal
-            // their ipma and move on. This is a sneaky way to provide iprp deduplication.
-
-            avifBool foundPreviousCell = AVIF_FALSE;
-            for (uint32_t dedupIndex = 0; dedupIndex < itemIndex; ++dedupIndex) {
-                avifEncoderItem * dedupItem = &encoder->data->items.item[dedupIndex];
-                if (item->dimgFromID == dedupItem->dimgFromID) {
-                    // We've already written dedup's items out. Steal their ipma indices and move on!
-                    memcpy(&item->ipma, &dedupItem->ipma, sizeof(struct ipmaArray));
-                    foundPreviousCell = AVIF_TRUE;
-                    break;
-                }
-            }
-            if (foundPreviousCell) {
+    {
```

Comment:
It does not seem necessary to put this block of code inside `{ }`. It seems that the purpose is to clearly limit the scope of the variable `dedup`? If so, an alternative is to set `dedup` to `NULL` after the `avifItemPropertyDedupDestroy(dedup)` call.

Reply:
I'll remove the block and set `dedup` to `NULL` instead.